extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -73,15 +74,30 @@ export const popupMessageListener = (request: Request) => {
   });
   keyManager.registerEncrypter(KeyManagerPlugins.ScryptEncrypter);
 
-  const _unlockKeystore = ({ password }: { password: string }) =>
-    keyManager.loadKey(localStorage.getItem(KEY_ID) || "", password);
+  const _unlockKeystore = ({
+    password,
+    keyID,
+  }: {
+    password: string;
+    keyID: string;
+  }) => keyManager.loadKey(keyID, password);
+
+  // this returns the first non hardware wallet (Hw) keyID, if it exists.
+  // Used for things like checking a password when a Hw is active.
+  const _getNonHwKeyID = () => {
+    const keyIdList = getKeyIdList();
+    const nonHwKeyIds = keyIdList.filter(
+      (k: string) => k.indexOf(HW_PREFIX) === -1,
+    );
+    return nonHwKeyIds[0] || "";
```

Comment:
I think this is good as it captures any weird edge case I'm not thinking of - but is there any reason to believe the first keyId in this list (the one the user onboards with) wouldn't always be "Non-HW" 🤔

Reply:
that's a good point I hadn't thought of that 🤔 

I think I'm in favor of this way, since it's more conservative. I also can't think of a way the first key won't always be non-hw, but in case that changes down the road, or if something funky happens with the list this method will still hold true